fireplace/dsl/random_picker.py

```diff
@@ -1,79 +1,97 @@
 import random
 from hearthstone.enums import CardType, Race, Rarity
 from .lazynum import LazyValue
+from copy import copy
+from .. import cards
 
 
 class RandomCardPicker(LazyValue):
 	"""
 	Store filters and generate a random card matching the filters on pick()
+	Constructor takes a single global set of filters and uses a default weighting of 1
+	Additional weighted filter sets can be added with add(), these will be merged with the global filters
 	"""
-	def __init__(self, *args, **filters):
-		self.args = args
-		self.filters = filters
+	def __init__(self, **filters):
+		self.weights = []
+		self.weightedfilters = []
+		self.globalfilter = filters
 		self.count = 1
-		self._cards = None
-		self.lazy_filters = False
-		for v in filters.values():
-			if isinstance(v, LazyValue):
-				self.lazy_filters = True
-				break
 
 	def __repr__(self):
-		return "%s(%r)" % (self.__class__.__name__, self.filters)
+		return "%s(%r)" % (self.__class__.__name__, self.globalfilter)
```

Comment:
Clarity. There is no longer a single set of filters, there is a main set of filters (globalfilters) and all of the weighted filters to which globalfilters is also applied individually.